common/src/test/scala/uk/ac/wellcome/message/MessageWriterTest.scala

```diff
@@ -0,0 +1,150 @@
+package uk.ac.wellcome.message
+
+import io.circe.Json
+import org.scalatest.concurrent.{IntegrationPatience, ScalaFutures}
+import org.scalatest._
+import uk.ac.wellcome.models.Sourced
+import uk.ac.wellcome.models.aws.SNSConfig
+import uk.ac.wellcome.test.fixtures._
+import uk.ac.wellcome.models.aws.S3Config
+import uk.ac.wellcome.s3.{KeyPrefixGenerator, S3ObjectLocation, S3ObjectStore}
+import uk.ac.wellcome.utils.JsonUtil._
+
+import scala.util.Success
+import uk.ac.wellcome.sns.SNSWriter
+
+class MessageWriterTest
+    extends FunSpec
+    with ScalaFutures
+    with Matchers
+    with Messaging
+    with IntegrationPatience
+    with Inside {
+
+  it(
+    "sends a message and returns a publish attempt with the id of the request") {
+    withLocalSnsTopic { topic =>
+      withLocalS3Bucket { bucket =>
+        val s3Config = S3Config(bucketName = bucket.name)
+
+        val snsConfig = SNSConfig(topic.arn)
+        val snsWriter = new SNSWriter(snsClient, snsConfig)
+
+        val s3ObjectStore = new S3ObjectStore[ExampleObject](
+          s3Client,
+          s3Config,
+          keyPrefixGenerator)
+
+        val messageWriter =
+          new MessageWriter[ExampleObject](snsWriter, s3Config, s3ObjectStore)
+
+        val message = ExampleObject("Some value")
+        val subject = "sns-writer-test-subject"
+
+        val eventualAttempt = messageWriter.write(message, subject)
+
+        whenReady(eventualAttempt) { pointer =>
+          val messages = listMessagesReceivedFromSNS(topic)
+          messages should have size (1)
+          messages.head.subject shouldBe subject
+
+          val pointer = fromJson[MessagePointer](messages.head.message)
+
+          inside(pointer) {
```

Comment:
I think it would be nicer to deconstruct without using inside. We need to access the bucketName and the key anyway to be able to retrieve the object and assert on it